conda_smithy/lint_recipe.py

```diff
@@ -42,14 +58,19 @@ def lintify(meta, recipe_dir=None):
         if not a_test_file_exists:
             lints.append('The recipe must have some tests.')
 
-    # 5: License cannot be 'unknown.'
-    license = meta.get('about', {}).get('license', '').lower()
-    if 'unknown' == license.strip():
-        lints.append('The recipe license cannot be unknown.')
+    # 5: Must have a valid OSI license.
+    license = meta.get('about', {}).get('license', '')
+    known = _parse_osi()
+    if license not in known:
+        suggestions = ' '.join(list(difflib.get_close_matches(license, known)))
+        if suggestions:
+            msg = '\nInstead of {} did you mean any of these: {}?'.format
+            msg = msg(license, suggestions)
+            lints.append(msg)
```

Comment:
The suggestion should be part of the lint itself. I've been meaning to make these lints a little more context aware (rather than just strings) so that we can have things like line numbers etc., suggestions are just another part of that IMHO. In the meantime, just concatenate your string to the end of the lint.